subprojects/core/src/main/java/org/gradle/execution/taskgraph/DefaultTaskExecutionGraph.java

```diff
@@ -216,7 +218,10 @@ public boolean hasTask(String path) {
 
     public List<Task> getAllTasks() {
         ensurePopulated();
-        return taskExecutionPlan.getTasks();
+        if (allTasks == null) {
+            allTasks = ImmutableList.copyOf(taskExecutionPlan.getTasks());
+        }
+        return allTasks;
```

Comment:
I don't think caching this is a good idea. The plan can get recalculated after `addTasks()` is called. We also don't call this method a lot, so it's probably not worth caching in the first place.

Reply:
Good point. I fixed the problem by resetting the cache in `ensurePopulated()`.